activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1108,6 +1108,10 @@ def arel_columns(columns)
     end
 
     def reverse_sql_order(order_query)
+      if offset_index != 0
+        raise IrreversibleOrderError, "Relations with offset cannot be reversed automatically."
```

Comment:
That makes sense. It triggers a deprecation warning, rather than raising an error, as of af79d7a below.